pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java

```diff
@@ -1748,4 +1751,46 @@ private void assertInvalidRetentionPolicyAsPartOfAllPolicies(Policies policies,
             assertTrue(e.getMessage().startsWith("Invalid retention policy"));
         }
     }
+
+    @Test
+    public void testExistNamespaceResourceGroup() throws PulsarAdminException {
+        String namespace = BrokerTestUtil.newUniqueName(this.testTenant + "/namespace");
+        ResourceGroup testResourceGroup = new ResourceGroup();
+        testResourceGroup.setDispatchRateInBytes(10000L);
+        testResourceGroup.setDispatchRateInMsgs(100);
+        testResourceGroup.setPublishRateInMsgs(100);
+        testResourceGroup.setPublishRateInBytes(10000L);
+        final String resourceGroupName = "test-resource-group";
+        admin.resourcegroups().createResourceGroup(resourceGroupName, testResourceGroup);
+        admin.namespaces().createNamespace(namespace);
+        admin.namespaces().setNamespaceResourceGroup(namespace, resourceGroupName);
+        String namespaceResourceGroup = admin.namespaces().getNamespaceResourceGroup(namespace);
+        assertEquals(namespaceResourceGroup, resourceGroupName);
+        admin.namespaces().removeNamespaceResourceGroup(namespace);
+        assertTrue(StringUtils.isBlank(admin.namespaces().getNamespaceResourceGroup(namespace)));
+    }
+
+    @Test
+    public void testNonExistNamespaceResourceGroup() throws PulsarAdminException {
+        String namespace = BrokerTestUtil.newUniqueName(this.testTenant + "/namespace");
+        ResourceGroup testResourceGroup = new ResourceGroup();
+        testResourceGroup.setDispatchRateInBytes(10000L);
+        testResourceGroup.setDispatchRateInMsgs(100);
+        testResourceGroup.setPublishRateInMsgs(100);
+        testResourceGroup.setPublishRateInBytes(10000L);
+        final String resourceGroupName = "test-resource-group";
+        admin.resourcegroups().createResourceGroup(resourceGroupName, testResourceGroup);
+        assertNotFound(__ -> admin.namespaces().setNamespaceResourceGroup(namespace, resourceGroupName));
```

Comment:
```suggestion
        assertEquals(assertThrows(PulsarAdminException.class,
                () -> admin.namespaces().setNamespaceResourceGroup(namespace,
                        resourceGroupName)).getStatusCode(), Response.Status.NOT_FOUND.getStatusCode());
```

Reply:
``expectThrows`` is more suitable for us.